packages/legend-query-builder/src/stores/QueryBuilderStateBuilder.ts

```diff
@@ -608,40 +607,45 @@ export class QueryBuilderValueSpecificationProcessor
 export const processParameters = (
   parameters: VariableExpression[],
   queryBuilderState: QueryBuilderState,
+  preservedParameters?: Map<VariableExpression, ValueSpecification | undefined>,
```

Comment:
TODO: since this is generic logic, for processing parameters, perhaps, we should not say it's `preserved` or something like that